types/linkifyjs/linkifyjs-tests.ts

```diff
@@ -1,141 +1,163 @@
+            }
+            return href;
         }
-        return href;
-    }
-});
-
-linkifyHtml("Hey @dhh, check out issue #23", {
-    formatHref: {
-        mention(href) {
-            return "https://github.com" + href;
-        },
-        ticket(href) {
-            return (
-                "https://github.com/SoapBox/linkifyjs/issues/" +
-                href.substring(1)
-            );
+    });
+
+    linkifyHtml("Hey @dhh, check out issue #23", {
+        formatHref: {
+            mention(href) {
+                return "https://github.com" + href;
+            },
+            ticket(href) {
+                return (
+                    "https://github.com/SoapBox/linkifyjs/issues/" +
+                    href.substring(1)
+                );
+            }
         }
-    }
-});
+    });
 
-/* ignoreTags */
+    /* ignoreTags */
 
-linkifyHtml(
-    // tslint:disable-next-line:prefer-template
-    'Please ignore <script>var a = {}; a.com = "Hi";</script> \n' +
-        "but do <span>b.ca</span>",
-    {
-        ignoreTags: ["script", "style"]
-    }
-);
+    linkifyHtml(
+        // tslint:disable-next-line:prefer-template
+        'Please ignore <script>var a = {}; a.com = "Hi";</script> \n' +
+            "but do <span>b.ca</span>",
+        {
+            ignoreTags: ["script", "style"]
+        }
+    );
 
-/* nl2br */
+    /* nl2br */
 
-/* tagName */
+    /* tagName */
 
-linkifyHtml("github.com", {
-    tagName: "span"
-});
+    linkifyHtml("github.com", {
+        tagName: "span"
+    });
 
-linkifyHtml("#swag", {
-    tagName: {
-        hashtag: "span"
-    }
-});
+    linkifyHtml("#swag", {
+        tagName: {
+            hashtag: "span"
+        }
+    });
 
-/* target */
+    /* target */
 
-linkifyHtml("github.com", {
-    target: "_parent"
-});
+    linkifyHtml("github.com", {
+        target: "_parent"
+    });
 
-linkifyHtml("test-email@example.com", {
-    target: {
-        url: "_parent",
-        email: null
-    }
-});
+    linkifyHtml("test-email@example.com", {
+        target: {
+            url: "_parent",
+            email: null
+        }
+    });
 
-/* validate */
+    /* validate */
 
-// Don't linkify links that don't begin in a protocol
-// e.g., "http://google.com" will be linkified, but "google.com" will not.
-linkifyHtml("www.google.com", {
-    validate: {
-        url(value) {
-            return /^(http|ftp)s?:\/\//.test(value);
+    // Don't linkify links that don't begin in a protocol
+    // e.g., "http://google.com" will be linkified, but "google.com" will not.
+    linkifyHtml("www.google.com", {
+        validate: {
+            url(value) {
+                return /^(http|ftp)s?:\/\//.test(value);
+            }
         }
-    }
+    });
+});
+
+describe("linkifyjs/react", () => {
+    it("should render a Linkify component", () => {
+        const rootElement = document.createElement("div");
+        ReactDOM.render(
```

Comment:
Can you add tests to test the entire breadth of documented use cases?

You can just rename this file to `.tsx` and copy all two use cases from this file:
https://soapbox.github.io/linkifyjs/docs/linkify-react.html

Afterwards you can add a comment so the the source of the test is clear! Something like:

```tsx
 // From the docs here: https://soapbox.github.io/linkifyjs/docs/linkify-react.html
```

After you add these tests, you will notice that you forgot to declare a prop!

Reply:
> After you add these tests, you will notice that you forgot to declare a prop!

`tagName` is an optional prop, it defaults to a span element. No real point in testing that if these files are checked by tsc, to be honest, there is no way to screw that up.

Reply:
I can add more tests but there is no browser (or jsdom) present so, now that I think of it, I'm really not sure how this code runs at all. I'll do some investigating.

Reply:
Ah I see what you mean by `tagName`, thanks 👍

Reply:
The tests are only compiled, they're not actual unit-tests, you were right. I will remove the calls to it. The describe blocks I think are fine as they add a bit of documentation.